### PR TITLE
[DOC]: Improve documentation

### PR DESCRIPTION
**Pull request by synchon**
* [x] description of feature/fix
* [x] tests added/passed
* [x] add an entry for the latest changes

This PR improves the documentation and in particular introduces space transformation via `SpaceWarper` and adds information about creating custom Preprocessors.

### CODE REVIEW

File: docs/extending/dependencies.rst [hunk 1]
```diff
@@ -0,0 +1,141 @@
+.. include:: ../links.inc
+
+.. _specifying_dependencies:
+
+Specifying dependencies
```
Comment (LeSasse) [inline docs/extending/dependencies.rst:5]: Should dependencies be capitalised?

File: docs/extending/dependencies.rst [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        ]
+
+        def __init__(
+            self, using: str, reference: str, on: Union[List[str], str]
+        ) -> None:
+            # validation and setting up
+
+
+Here, you see a new class attribute ``_CONDITIONAL_DEPENDENCIES`` which is a
+list of dictionaries having two keys:
```
Comment (LeSasse) [inline docs/extending/dependencies.rst:99]: "having two keys" -> "with two keys"

File: docs/extending/dependencies.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+* ``using`` (str) : lowercased name of the toolbox
+* ``depends_on`` (object) : a class which implements the particular tool's use
+
+It is mandatory to have the ``using`` positional argument in the constructor in
+this case as the validation starts with this and moves further. It is also
+mandatory to only allow the value of ``using`` argument to be one of them
+specified in the ``using`` key of ``_CONDITIONAL_DEPENDENCIES`` entries.
+
+For brevity, we only show the ``FSLWarper`` here but ``ANTsWarper`` looks very
```
Comment (LeSasse) [inline docs/extending/dependencies.rst:109]: ``ANTsWarper``` the capitalisation here makes me anxious, but I'll allow it

File: docs/extending/parcellations.rst [hunk 4]
```diff
@@ -52,13 +53,13 @@ first label in this list corresponds to the first integer label in the
...
+``"MNI152NLin2009cAsym"`` or ``"native"`` (scanner-native space).
 
 Step 1: Prepare code to register a parcellation
 -----------------------------------------------
 
-Now we know everything that we need to know to make sure junifer can use our
-own parcellation to compute any parcellation-based marker. For example,
+Now we know everything that we need to know to make sure ``junifer`` can use our
+own parcellation to compute any parcellation-based Marker. For example,
 a simple example could look like this:
```
Comment (LeSasse) [inline docs/extending/parcellations.rst:63]: "For example, a simple example could look like this:" -> "A simple example could look like this:"

File: docs/extending/preprocessor.rst [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+        self,
+        input: dict[str, Any],
+        extra_input: dict[str, Any] | None = None,
+    ) -> tuple[dict[str, Any], dict[str, Any] | None]:
+         input["data"] = nimg.smooth_img(imgs=input["data"], fwhm=self.fwhm)
+         return input, None
+
+     ...
+
+Step 4: Finalise the Preprocessor
```
Comment (LeSasse) [inline docs/extending/preprocessor.rst:160]: I believe in multiple places I saw the american spelling of things, so this should be Finalize. Can we record in some central place i.e. something like "How to contribute to docs" that we aim to use the american spelling?

**Review by LeSasse — CHANGES_REQUESTED** [review]
Overall very strong improvements, I only have a few minor comments and questions.

File: docs/builtin.rst [hunk 6]
```diff
@@ -122,6 +122,54 @@ Planned
      - :gh:`47`
 
 
+Preprocessor
```
[the remark this answers is not in the document]
Comment (synchon) [reply docs/builtin.rst:125]: Yeah it does. `preprocess` accepts a list now and the execution follows the sequence you specify in the YAML.

File: docs/builtin.rst [hunk 7]
```diff
@@ -277,7 +321,7 @@ Available
    * - Name
      - Options
      - Keys
-     - Spaces
+     - Template spaces
      - Version added
```
[the remark this answers is not in the document]
Comment (synchon) [reply docs/builtin.rst:325]: Yeah missed it, good catch.

File: docs/builtin.rst [hunk 8]
```diff
@@ -663,14 +707,14 @@ Available
...
      - Adapts to the target data
      - 0.0.2
-     - | Compute the whole-brain mask. This mask is calculated using
-       | MNI152 1mm-resolution template mask onto the target image.
-       | See :func:`nilearn.masking.compute_brain_mask`
-   * - Nilearn's mask computed from FMRI data
+     - | Compute the whole-brain, gray-matter or white-matter mask using
+       | the template and the resolution from the target image. The
+       | templates are obtained via ``templateflow``.
+   * - Nilearn's mask computed from fMRI data
```
[the remark this answers is not in the document]
Comment (synchon) [reply docs/builtin.rst:717]: That would be better, fair point.

File: docs/extending/dependencies.rst (hunk 3 above)
Comment (synchon) [reply docs/extending/dependencies.rst:109]: Its to follow the convention of the tool name like in other places in the code base.

Comment (LeSasse) [reply docs/extending/dependencies.rst:109]: I understand :)

File: docs/extending/preprocessor.rst (hunk 5 above)
Comment (LeSasse) [reply docs/extending/preprocessor.rst:160]: Ok, i will point out the american way when i find them.

File: docs/extending/datagrabber.rst [hunk 9]
```diff
@@ -339,7 +356,7 @@ method, in the same order.
       return ["subject", "session"]
 
 
-So, to summarize, our datagrabber will look like this:
+So, to summarize, our DataGrabber will look like this:
```
Comment (LeSasse) [inline docs/extending/datagrabber.rst:359]: this is american spelling

Comment (synchon) [reply docs/extending/datagrabber.rst]: Boston tea party reversed.

File: docs/using/codeless.rst [hunk 10]
```diff
@@ -5,7 +5,7 @@
 Code-less Configuration
 =======================
 
-On of the most important features of junifer is its capacity to run without
+On of the most important features of ``junifer`` is its capacity to run without
```
Comment (fraimondo) [inline docs/using/codeless.rst:8]: "One of..."

**Review by fraimondo — APPROVED** [review]
Just a typo, for the rest, I'm good.

**Pull request merged**